pkg/provisioner/ironic/clients/client_test.go

```diff
@@ -0,0 +1,230 @@
+)
+
+func TestIronicClient(t *testing.T) {
+	type args struct {
+		ironicEndpoint string
+		auth           AuthConfig
+		tls            TLSConfig
+	}
+
+	badClientArgs := args{
+		ironicEndpoint: "test",
+		auth: AuthConfig{
+			Type:     "unsupportedTYpe",
+			Username: "username",
+			Password: "password",
+		},
+		tls: TLSConfig{
+			TrustedCAFile:         "",
+			ClientCertificateFile: "",
+			ClientPrivateKeyFile:  "",
+			InsecureSkipVerify:    true,
+			SkipClientSANVerify:   true,
+		},
+	}
+
+	noAuthClientArgs := args{
+		ironicEndpoint: "http://localhost",
+		auth: AuthConfig{
+			Type:     NoAuth,
+			Username: "",
+			Password: "",
+		},
+		tls: TLSConfig{
+			TrustedCAFile:         "",
+			ClientCertificateFile: "",
+			ClientPrivateKeyFile:  "",
+			InsecureSkipVerify:    true,
+			SkipClientSANVerify:   true,
+		},
+	}
+	noAuthClient, _ := IronicClient(noAuthClientArgs.ironicEndpoint, noAuthClientArgs.auth, noAuthClientArgs.tls)
+
+	noAuthTlsClientArgs := args{
+		ironicEndpoint: "https://localhost",
+		auth: AuthConfig{
+			Type:     NoAuth,
+			Username: "",
+			Password: "",
+		},
+		tls: TLSConfig{
+			TrustedCAFile:         "/path/to/ca/file.crt",
+			ClientCertificateFile: "/path/to/cert/file.crt",
+			ClientPrivateKeyFile:  "/path/to/cert/file.key",
+			InsecureSkipVerify:    true,
+			SkipClientSANVerify:   true,
+		},
+	}
+	noAuthTlsClient, _ := IronicClient(noAuthTlsClientArgs.ironicEndpoint, noAuthTlsClientArgs.auth, noAuthTlsClientArgs.tls)
+
+	basicAuthClientArgs := args{
+		ironicEndpoint: "http://localhost",
+		auth: AuthConfig{
+			Type:     HTTPBasicAuth,
+			Username: "username",
+			Password: "password",
+		},
+		tls: TLSConfig{
+			TrustedCAFile:         "",
+			ClientCertificateFile: "",
+			ClientPrivateKeyFile:  "",
+			InsecureSkipVerify:    true,
+			SkipClientSANVerify:   true,
+		},
+	}
+	basicAuthClient, _ := IronicClient(basicAuthClientArgs.ironicEndpoint, basicAuthClientArgs.auth, basicAuthClientArgs.tls)
+
+	basicAuthTlsClientArgs := args{
+		ironicEndpoint: "https://localhost",
+		auth: AuthConfig{
+			Type:     HTTPBasicAuth,
+			Username: "username",
+			Password: "password",
+		},
+		tls: TLSConfig{
+			TrustedCAFile:         "/path/to/ca/file.crt",
+			ClientCertificateFile: "/path/to/cert/file.crt",
+			ClientPrivateKeyFile:  "/path/to/cert/file.key",
+			InsecureSkipVerify:    true,
+			SkipClientSANVerify:   true,
+		},
+	}
+	basicAuthTlsClient, _ := IronicClient(basicAuthTlsClientArgs.ironicEndpoint, basicAuthTlsClientArgs.auth, basicAuthTlsClientArgs.tls)
+
+	tests := []struct {
+		name       string
+		args       args
+		wantClient *gophercloud.ServiceClient
+		wantErr    bool
+	}{
+		{
+			name:       "non supported auth type return error",
+			args:       badClientArgs,
+			wantClient: nil,
+			wantErr:    true,
+		},
+		{
+			name:       "noauth auth type without tls return no error",
+			args:       noAuthClientArgs,
+			wantClient: noAuthClient,
+			wantErr:    false,
```

Comment:
Let's maybe make this a string and check that `err.String()` contains it? This way we rule out false positives because of unrelated errors.

Reply:
Updated